components/layout_thread/lib.rs

```diff
@@ -1080,10 +1097,11 @@ impl LayoutThread {
 
         let element = match document.root_element() {
             None => {
+                self.layout_query_timestamp = data.reflow_goal.timestamp().expect("Not a QueryMsg");
                 // Since we cannot compute anything, give spec-required placeholders.
                 debug!("layout: No root node: bailing");
                 match data.reflow_goal {
-                    ReflowGoal::LayoutQuery(ref quermsg, _) => match quermsg {
+                    ReflowGoal::LayoutQuery(ref query_msg, _) => match query_msg {
```

Comment:
What if we do this instead of adding the timestamp method to ReflowGoal?
```rust
ReflowGoal::LayoutQuery(ref query_msg, timestamp) => {
    self.layout_query_timestamp = timestamp;
    match query_msg {
```

Reply:
Also, this only stores a timestamp in the case where there is no root element. We also care about the case where there is a root element, so we can do this instead outside of this match:
```rust
if let ReflowGoal::LayoutQuery(_, timestamp) = data.reflow_goal {
    self.layout_query_timestamp = timestamp;
}
```